### PR TITLE
feat: separate out immutables from RouterImmutables

### PR DESCRIPTION
**Pull request by zhongeric**
This will allow for the individual modules to be exported and used in other projects

### CODE REVIEW

File: contracts/modules/uniswap/UniswapImmutables.sol [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    /// @dev The UniswapV3Pool initcodehash
+    bytes32 internal immutable UNISWAP_V3_POOL_INIT_CODE_HASH;
+
+    constructor(UniswapParameters memory params) {
+        UNISWAP_V2_FACTORY = params.v2Factory;
+        UNISWAP_V2_PAIR_INIT_CODE_HASH = params.pairInitCodeHash;
+        UNISWAP_V3_FACTORY = params.v3Factory;
+        UNISWAP_V3_POOL_INIT_CODE_HASH = params.poolInitCodeHash;
+    }
+}
```
Comment (ewilz) [inline contracts/modules/uniswap/UniswapImmutables.sol:30]: probably wanna add an extra line to this file (me editor does this automatically?)

File: contracts/modules/uniswap/v3/V3SwapRouter.sol [hunk 2]
```diff
@@ -9,14 +9,15 @@ import {IUniswapV3SwapCallback} from '@uniswap/v3-core/contracts/interfaces/call
...
 import {Constants} from '../../../libraries/Constants.sol';
 import {ERC20} from 'solmate/src/tokens/ERC20.sol';
 
 /// @title Router for Uniswap v3 Trades
-abstract contract V3SwapRouter is RouterImmutables, Permit2Payments, IUniswapV3SwapCallback {
+abstract contract V3SwapRouter is UniswapImmutables, Permit2Payments, IUniswapV3SwapCallback {
     using V3Path for bytes;
     using BytesLib for bytes;
-    using SafeCast for uint256;
+    using SafeCast for uint256;
```
Comment (ewilz) [inline contracts/modules/uniswap/v3/V3SwapRouter.sol:20]: again..just looks like a linter needs to run

**Review by ewilz — APPROVED** [review]
lgtm! think I like this better anyway. I suppose the constructor could also just take 3 structs and hand them off respectively. But not feeling too opinionated on that..

**Comment by zhongeric** [comment]
Yea I thought about that, didn't really want to have to refactor all of the tests though haha

**Pull request merged**